### PR TITLE
[pytest/default_route_check]: check various aspect related to default route learnt from bgp

### PR DESCRIPTION
**Pull request by lguohan**
Signed-off-by: Guohan Lu <gulv@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)

#### How did you verify/test it?
```
johnar@1cafe0c9f994:/data/tests$ py.test $PYTEST_COMMON_OPTS --log-file logs/abc.log route/test_default_route.py  | tee ~/abc.txt               ============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-4.6.9, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
ansible: 2.8.7
rootdir: /data/tests, inifile: pytest.ini
plugins: ansible-2.2.2
collecting ... collected 2 items

route/test_default_route.py::test_default_route_set_src PASSED           [ 50%]
route/test_default_route.py::test_default_ipv6_route_next_hop_global_address PASSED [100%]

=============================== warnings summary ===============================
```

### CODE REVIEW

File: tests/route/test_default_route.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    lo_ipv4 = None
+    lo_ipv6 = None
+    los = config_facts.get("LOOPBACK_INTERFACE", {})
+    logger.info("Loopback IPs: {}".format(los))
+    for k, v in los.items():
+        if k == "Loopback0":
+            for ipstr in v.keys():
+                ip = ipaddress.ip_interface(ipstr)
+                if ip.version == 4:
+                    lo_ipv4 = ip
```
Comment (yxieca) [inline tests/route/test_default_route.py:37]: Should lo_ipv4 and lo_ipv6 be a list to accommodate multiple loopback addresses?

File: tests/route/test_default_route.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    if lo_ipv6 is None:
+        pytest.fail("cannot find ipv6 Loopback0 address")
+
+    rt = duthost.command("ip route list match 0.0.0.0")['stdout_lines']
+    m = re.match(r"^default proto bgp src (\S+)", rt[0])
+    if m:
+        if ipaddress.ip_address(m.group(1)) != lo_ipv4.ip:
+            pytest.fail("default route set src to wrong IP {} != {}".format(m.group(1), lo_ipv4.ip))
+    else:
+        pytest.fail("default route do not have set src. {}".format(rt))
```
Comment (yxieca) [inline tests/route/test_default_route.py:52]: You could use https://github.com/Azure/sonic-mgmt/blob/master/tests/common/helpers/assertions.py#L3 to simplify code.

File: tests/route/test_default_route.py (hunk 1 above)
Comment (lguohan) [reply tests/route/test_default_route.py:37]: no. not heeded.  two routes only requires two src ip

File: tests/route/test_default_route.py (hunk 2 above)
Comment (lguohan) [reply tests/route/test_default_route.py:52]: using if condition; then fail works good here. if you prefer, I can add a pytest_fail wrapper in your helper class and use it.

Comment (yxieca) [reply tests/route/test_default_route.py:52]: We can improve later.

**Pull request merged**